src/library/basetracktablemodel.cpp

```diff
@@ -157,9 +121,28 @@ BaseTrackTableModel::BaseTrackTableModel(
 void BaseTrackTableModel::initTableColumnsAndHeaderProperties(
         const QStringList& tableColumns) {
     m_columnCache.setColumns(tableColumns);
-    if (m_columnHeaders.size() < tableColumns.size()) {
-        m_columnHeaders.resize(tableColumns.size());
+
+    // Reset the column headers.
+    m_columnHeaders.clear();
+    m_columnHeaders.resize(tableColumns.size());
+
+    // Init the mapping of all columns, even for internal columns that are
+    // hidden/invisible. Otherwise mapColumn() would not return a valid result
+    // for those columns.
+    for (int columnValue = 0; columnValue < ColumnCache::NUM_COLUMNS; ++columnValue) {
+        const auto column = static_cast<ColumnCache::Column>(columnValue);
+        DEBUG_ASSERT(column != ColumnCache::COLUMN_LIBRARYTABLE_INVALID);
+        const int headerIndex = m_columnCache.fieldIndex(column);
+        if (headerIndex < 0) {
+            // Missing table column.
+            continue;
+        }
+        DEBUG_ASSERT(headerIndex < m_columnHeaders.size());
+        m_columnHeaders[headerIndex].column = column;
+        DEBUG_ASSERT(mapColumn(headerIndex) == column);
     }
```

Comment:
It turns out that the whole loop Is redundant, because the header properties are reinitialized below. There is only a dummy entry missing rot the ID header. Something like:

```
    setHeaderProperties(
            ColumnCache::COLUMN_LIBRARYTABLE_ID,
            QString(), // hidden
            0);
```

You may assert below `initHeaderProperties()` that all `m_columnHeaders` have been initialized. 

It would be better to not initalizer the header properties at runtime, because all columns are known at compile time. But it is OK for me to not extend the scope of this PR as suggested in the other comment.

Reply:
That's how an alternative initialization ordering looks like: https://github.com/mixxxdj/mixxx/pull/13682/commits/e0d43748aad2197f7ce6b651db585c30a0627708. IMHO more complicated and cluttered.

I suppose this code either vanishes or needs to be rewritten when migrating to QML. I wouldn't put too much effort in it.

Squashing can be done on merge. The PR title already contains the final commit message.

Reply:
I was probably not clear enough: Please **remove** this complicated and cluttered loop. It is not necessary. 

Instead just this is requied: 
```
 setHeaderProperties(
            ColumnCache::COLUMN_LIBRARYTABLE_ID,
            QString(), // hidden
            0);
```